createdisk-library.sh

```diff
@@ -189,11 +189,24 @@ function install_additional_packages() {
         # SCP the downloaded rpms to VM
         ${SCP} -r ${pkgDir}/packages core@${vm_ip}:/home/core/
 
+        # Create local repo of downloaded RPMs in the VM
+        ${SSH} core@${vm_ip} 'sudo bash -x -s' <<EOF
+            podman run --rm -v /home/core/packages:/packages:Z quay.io/centos/centos:stream9 sh -c "dnf install -y createrepo && createrepo /packages"
+            podman rmi quay.io/centos/centos:stream9
+EOF
+        ${SSH} core@${vm_ip} "sudo bash -c 'cat > /etc/yum.repos.d/local.repo << EOF
+[local]
+name=Local repo
+baseurl=file:///home/core/packages/
+enabled=1
+gpgcheck=0
+EOF'"
```

Comment:
Have you considered serving this repo from the host? this might be a bit simpler, no need to copy files between host and guest, possibly less cleanup to do, ...

Reply:
No, but except scp packages it would be same steps even on host and then another step to run a local server on host which can serve it to guest.